### PR TITLE
WIP - Write Test Suite Docs (Fixes #178)

### LINKED ISSUE #178: Write documentation for test suite
This includes how to contribute, as detailed in #189.

### PR DESCRIPTION
**Pull request by hayleyavw**
Struggling to flesh this out, input would be good :)

### CODE REVIEW

File: docs/source/developer/test_suite.rst [hunk 1]
```diff
@@ -4,32 +4,80 @@ Test Suite
 .. contents:: Contents
   :local:
 
+
+Running the Test Suite
+==============================================================================
+
+``./csu dev test`` will run the entire test suite. For a full list of test commands available, see :doc:`../getting_started/helper_commands`.
```
Comment (JackMorganNZ) [inline docs/source/developer/test_suite.rst]: Might be worth changing 'a full list of test commands available' to 'running a specific test suite, viewing code coverage, and more testing commands'.

File: docs/source/developer/test_suite.rst [hunk 2]
```diff
@@ -4,32 +4,80 @@ Test Suite
...
+
+
 Structure
 ==============================================================================
 
-.. code-block:: none
+The ``tests/`` directory is in the ``csunplugged/`` directory (at the same
+level as the apps) and is structured as follows:
```
Comment (JackMorganNZ) [inline docs/source/developer/test_suite.rst]: We should be explicit stating that all tests are located here, and not within their application directories.

File: docs/source/developer/test_suite.rst [hunk 3]
```diff
@@ -4,32 +4,80 @@ Test Suite
...
+Adding Tests
+==============================================================================
+
+When writing a new test function, it is important that the method name is as 
+descriptive as possible. The method name should also be prefixed with ``test_``
+as the test suite will only execute methods with this prefix.
+
+.. note::
+
+  We use `Codecov`_ to check the coverage of our tests. Every Pull Request should
```
Comment (JackMorganNZ) [inline docs/source/developer/test_suite.rst]: Doesn't need to specifically increase coverage, but should cover 100% of the difference (which therefore increases coverage).

File: docs/source/developer/test_suite.rst [hunk 4]
```diff
@@ -4,32 +4,80 @@ Test Suite
...
+  increase the overall coverage, Travis will fail if this is not the case.
+
+
+.. _Codecov: https://codecov.io/gh
```
Comment (JackMorganNZ) [inline docs/source/developer/test_suite.rst]: Should be https://codecov.io/

**Comment by hayleyavw** [comment]
Does this page need anything more before we can add it? Not really sure what to add, it might just have to be worked on overtime as we change things?

**Comment by JackMorganNZ** [comment]
This looks fine for the moment, and it should be adjusted over time.

**Pull request merged**